src/app/state/tabs/sql.rs

```diff
@@ -126,34 +158,64 @@ impl<'app> SQLTabState<'app> {
     }
 
     pub fn update_editor_content(&mut self, key: KeyEvent) {
-        self.editor.input(key);
+        match self.mode {
+            SQLTabMode::Normal => self.editor.input(key),
+            SQLTabMode::DDL => self.ddl_editor.input(key),
+        };
```

Comment:
hate that i have to do this in all these methods. i tried adding `active_editor_mut` method but was running into lifetime issues with `TextArea`.  Ive had issues with lifetimes from that a couple times i need to revisit this at some point to see if there is a way to improve the situation.